core/client/kt/verify_test.go

```diff
@@ -17,56 +17,106 @@ package kt
 )
 
-func TestVerifyCommitment(t *testing.T) {
-	commitment, committed, err := commitments.Commit(primaryUserID, primaryAppID, profileData)
+// Test vectors were obtained by observing the integration tests.
+func TestVerifyGetEntyrResponse(t *testing.T) {
+	ctx := context.Background()
+	vrfPub, err := p256.NewVRFVerifierFromPEM(VRFPub)
 	if err != nil {
-		t.Fatalf("Commit(%v, %v)=%v", primaryUserID, profileData, err)
+		t.Fatal(err)
 	}
-
-	entry := &tpb.Entry{Commitment: commitment}
-	validEntryData, err := proto.Marshal(entry)
+	mapPub, err := keys.NewFromPublicDER(MapPub)
 	if err != nil {
-		t.Fatalf("Marshal(%v)=%v", entry, err)
+		t.Fatal(err)
 	}
-	fakeEntryData := validEntryData[:len(validEntryData)-1]
-
-	// Create a dummy client verifier.
-	verifier := New(nil, nil, nil, nil)
+	v := New(vrfPub, maphasher.Default, mapPub, fake.NewFakeTrillianLogVerifier())
 	for _, tc := range []struct {
+		desc          string
+		wantErr       bool
 		userID, appID string
-		entryData     []byte
-		committed     *tpb.Committed
-		want          bool
+		trusted       *trillian.SignedLogRoot
+		in            *keytransparency_v1_types.GetEntryResponse
 	}{
-		{primaryUserID, primaryAppID, validEntryData, committed, false}, // Working case
-		{primaryUserID, primaryAppID, validEntryData, nil, false},       // nil committed
-		{primaryUserID, primaryAppID, fakeEntryData, committed, true},   // Unmarshable entry
-	} {
-		resp := &tpb.GetEntryResponse{
-			Committed: tc.committed,
-			LeafProof: &trillian.MapLeafInclusion{
-				Leaf: &trillian.MapLeaf{
-					LeafValue: tc.entryData,
+		{
+			desc:    "empty",
+			userID:  "noalice",
+			appID:   "app",
+			trusted: &trillian.SignedLogRoot{},
+			in: &keytransparency_v1_types.GetEntryResponse{
+				VrfProof:  []byte{0x8, 0xca, 0x1, 0xaa, 0x1, 0x74, 0x70, 0xf4, 0xbf, 0x96, 0x69, 0xff, 0x73, 0x5d, 0xc7, 0x9d, 0x2b, 0x51, 0xb0, 0xa5, 0x42, 0xb8, 0x8c, 0x42, 0x34, 0xfc, 0xa1, 0xa2, 0xc0, 0x80, 0xdf, 0x76, 0x33, 0xfc, 0x7d, 0xee, 0x4a, 0xf3, 0x18, 0xea, 0x30, 0xc4, 0xa4, 0x6f, 0x31, 0xc9, 0x20, 0x73, 0x42, 0x84, 0xd9, 0x71, 0x39, 0x52, 0xb2, 0x8f, 0x58, 0x52, 0x4, 0x53, 0x87, 0x8, 0x3e, 0x81, 0x4, 0xb, 0x13, 0x89, 0xd7, 0xc6, 0x63, 0x22, 0x39, 0x18, 0x73, 0x72, 0xfa, 0x32, 0xf6, 0xeb, 0x3, 0x8, 0x5d, 0x7, 0x4e, 0x2, 0x3a, 0xc6, 0x7f, 0x89, 0xe8, 0x44, 0x27, 0xcb, 0x73, 0xdc, 0xf2, 0x2f, 0xcc, 0xcd, 0x90, 0x6e, 0x97, 0xcb, 0x22, 0xff, 0x6e, 0xdb, 0x74, 0x22, 0xbf, 0x28, 0x27, 0x9b, 0x9e, 0x26, 0x1a, 0xe4, 0xc6, 0x16, 0x59, 0x4f, 0x7d, 0xcc, 0xb9, 0x8e, 0x7d, 0x41, 0xf7},
```

Comment:
That's true, but makes the test vectors harder to regenerate with `log.Printf("%#v")`